File: snippets/fsharp/System/String/Compare/string.comp4.fs
module string.comp4

open System
open System.Globalization

//<snippet1>
let symbol r =
    if r < 0 then "<"
    elif r > 0 then ">"
    else "="

let str1 = "change"
let str2 = "dollar"

let relation1 = 
    String.Compare(str1, str2, false, CultureInfo "en-US")
    |> symbol
printfn $"For en-US: {str1} {relation1} {str2}"

let relation2 = 
    String.Compare(str1, str2, false, CultureInfo "cs-CZ")
    |> symbol
printfn $"For cs-CZ: {str1} {relation2} {str2}"

(*
This example produces the following results.
For en-US: change < dollar
For cs-CZ: change > dollar
*)
//</snippet1>

File: snippets/csharp/System/String/Compare/string.comp4.cs
using System;
using System.Globalization;

class Sample0
{
    //<snippet1>
    public static void Main()
    {
        string str1 = "change";
        string str2 = "dollar";
        string relation;

        relation = symbol(string.Compare(str1, str2, false, new CultureInfo("en-US")));
        Console.WriteLine($"For en-US: {str1} {relation} {str2}");

        relation = symbol(string.Compare(str1, str2, false, new CultureInfo("cs-CZ")));
        Console.WriteLine($"For cs-CZ: {str1} {relation} {str2}");
    }

    private static string symbol(int r)
    {
        string s = "=";
        if (r < 0) s = "<";
        else if (r > 0) s = ">";
        return s;
    }

    /*
    This example produces the following results.
    For en-US: change < dollar
    For cs-CZ: change > dollar
    */

    //</snippet1>
}
